actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -151,6 +154,17 @@ def valid_accept_header
       def use_accept_header
         !self.class.ignore_accept_header
       end
+
+      private
+
+      def extract_variant!
+        if parameters.key?(:format)
+          segments = parameters[:format].to_s.split("+")
```

Comment:
No, the test isn't invalid.
But it suggests you may be hooking in at the wrong level.

Thinking out loud here it might be better to extend the routing parser. What are your thoughts on this, @pixeltrix? :smile: